### ISSUE TITLE
VisualDensity is inverted

### ISSUE THREAD

**Issue by Levi-Lesches** [flutter/flutter#87754]
From the description at [`ThemeData.visualDensity`](https://api.flutter.dev/flutter/material/ThemeData/visualDensity.html): 

> Density, in the context of a UI, is the vertical and horizontal "compactness" of the elements in the UI.

> ...negative values indicate a denser, more compact, UI, and positive values indicate a less dense, more expanded, UI.

> A larger value translates to a spacing increase (less dense), and a smaller value translates to a spacing decrease (more dense).

Is this not inconsistent? How can you say that "a larger value [of `visualDensity`] translates to a spacing increase (less dense)" and vice versa? A higher value for a variable called "density" should _increase_ density! I don't know how many Flutter widgets use this property already, but it should be sooner rather than later. 

I noticed that [material.io](https://material.io/design/layout/applying-density.html#components) gets this wrong too (emphasis mine):
> The density scale allows you to control the density of individual components when needed.
> 
> The density scale is numbered, starting at 0 for a component’s default density. The scale moves to negative numbers (-1, -2, -3) as space decreases, creating higher density. **_Each increment represents a decrease in the height of a component by 4dps._** 

Seems like both pages are really describing _spacing_, not density. 

## Proposal

Invert the definition of density such that higher values actually correlate to a higher density. 

Should I file an issue at https://github.com/material-components/material-components as well?

**Comment by HansMuller** [comment]
I don't think we could safely invert the meaning of visualDensity now because backwards compatibility
CC @gspencergoog

**Comment by gspencergoog** [comment]
Flutter's implementation is derived from the Material Design definition. I actually agree with your point, but my guess is that the horses have left the barn on this one.

**Comment by Levi-Lesches** [comment]
Feels like the horses are going the wrong way! I hope the material team responds sometime soon. Here's a demo of the effect of "increasing" density: https://dartpad.dev/?id=9512589f48184c0d944582b85fb57963.

Regarding backwards compatibility: I only found out about visual density through looking at the API for ListTile, and the `ListTile.dense` property is still there. Same for `Chip.padding`, `ButtonStyle.padding`, and more. And on [the material docs](https://material.io/design/layout/applying-density.html#implementation), Flutter is listed as "planned". A quick Google search for "flutter visual density" shows nothing except for API docs. All that to say that it may not be too late to make the switch before `VisualDensity` is used more widely.
